packages/core/src/external/aws/lambda-logic/bundle-to-brief.ts

```diff
@@ -1,128 +1,747 @@
-  // const prompt2 = `Today's date is ${todaysDate}. Given the attached medical documents, summarize them into a concise summary of the patients medical history. In your first sentence, include the patient's age/gender and all significant medical conditions. In the second sentence, summarize any notable recent health events (eg hospitalizations or emergency department visits) and any requested follow-up. Do not include directions on taking medications. Here is an example of an ideal summary: Patient is a 65 yo male with hx of HTN, poorly controlled diabetes, and smoking who presented with 3 hours of crushing substernal chest pain. They underwent a cardiac cath with stent placement and were discharged on aspirin and metoprolol.`;
-  const body = processedBundleInput;
-  try {
-    const brief = bedrockUtils.getBedrockResponse({
-      prompt,
-      body,
-    });
-    log(`Brief generated.`);
-    return brief;
-  } catch (error) {
-    log(`Error generating brief: ${errorToString(error)}`);
-    return undefined;
+  if (res.resourceType === "AllergyIntolerance") {
+    updRes.status = Array.from(
+      new Set(res.clinicalStatus?.coding?.flatMap(coding => coding.code || []))
+    ).join(", ");
+    if (updRes.status === "") delete updRes.status;
   }
-}
 
-export async function prepareBundleForBrief(bundle: Bundle): Promise<string | undefined> {
-  const bundleCopy = cloneDeep(bundle);
+  if (res.resourceType === "Immunization") {
+    delete res.lotNumber;
+
+    if (res.vaccineCode) {
+      const resVaccineCodeString = JSON.stringify(res.vaccineCode).toLowerCase();
+      if (
+        resVaccineCodeString.includes("no data") ||
+        resVaccineCodeString.includes("no immunization")
+      ) {
+        return undefined;
+      }
+
+      updRes.vaccineCode = getUniqueDisplays(res.vaccineCode);
+    }
+
+    delete res.doseQuantity;
+  }
+
+  if (res.resourceType === "Practitioner") {
+    const name = getNameString(res.name);
+    updRes.name = name;
+
+    const qualificationsText = getUniqueDisplays(res.qualification?.[0]?.code);
+    if (qualificationsText) updRes.qualification = qualificationsText;
+
+    updRes.address = getAddressString(res.address);
+  }
 
-  if (!bundle.entry?.length) return undefined;
-  const dedupedBundle = deduplicateBundleResources(bundleCopy);
-  if (!dedupedBundle.entry?.length) return undefined;
-  const filteredString = filterBundleResources(dedupedBundle);
-  if (!filteredString) return undefined;
-  return (await preprocessDataWithRag(filteredString)).trim();
+  if (res.resourceType === "Procedure") {
+    const name = getUniqueDisplays(res.code);
+    if (name) {
+      updRes.name = name;
+      if (name.includes("no data")) return undefined;
+    }
+
+    delete updRes.code;
+
+    if (updRes.status === "") delete updRes.status;
+
+    delete res.reasonCode; // TODO: Introduce term server lookup here
```

Comment:
tie TODOs with an issue #

Reply:
Added to #https://github.com/metriport/metriport-internal/issues/2510